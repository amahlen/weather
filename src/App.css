@import url('https://fonts.googleapis.com/css?family=Merriweather:100,200,300,300i,400,400i,700');
@import url('https://fonts.googleapis.com/css?family=Roboto+Slab:400,700');
@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400');

body {
  font-family: "Open Sans", serif;
}

.wrapper {/*styling the wrapper class*/
  background: linear-gradient(to right, #70bef4, #066ab0);
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.main {/*styling the main class*/
  height: 90vh;
  background: #fff;
  box-shadow: 0px 13px 40px -13px rgba(0,0,0,0.75);
  width: 80%;
  margin: 0 auto;
}

.title-container {/*styling the title-container class*/
  height: 90vh;
  background-size: cover;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  color: rgb(11, 3, 82);
}

.title-container__title {/*styling the tite-container_title class*/
  font-size: 50px;
  letter-spacing: 2px;
  line-height: 1.3;
  font-family: 'Roboto Slab', serif;
}

.title-container__subtitle {/*styling the tite-container_subtitle class*/
  font-style: italic;
  font-size: 25px;
  font-weight: 100;
  letter-spacing: 1px;
  line-height: 1.5;
  font-family: "Merriweather", serif;
}

.form-container {/*styling the form-container class*/
  background-color: lightblue; 
  height: 90vh;
  padding-top: 100px;
  padding-left: 50px;
}

input[type="text"] {/*styling the input*/
  background-color: transparent;
  border: 0;
  border-bottom: solid 1px #f16051;
  width: 30%;
  padding-bottom: 4px;
  color: #fff !important;
  font-weight: lighter;
  letter-spacing: 2px;
  margin-bottom: 30px;
  margin-right: 20px;
  font-size: 20px;
}

input[type="text"] { /*styling the input */
    outline: none;
}

input:-webkit-autofill {/*styling the input*/
    -webkit-box-shadow: 0 0 0 30px #2c3e50 inset;
    -webkit-text-fill-color: #fff !important;
}

button {/*styling the button*/
  border: 0;
  padding: 8px 20px;
  margin: 0 2px;
  border-radius: 2px;
  font-weight: lighter;
  letter-spacing: 1px;
  font-size: 15px;
  cursor: pointer;
  background-color: #49aee9;
  color: #fff;
  font-weight: 100;
}

button:active {
  outline: none;
}

.weather__info {/*styling the weather__info class*/
  width: 60%;
  font-size: 20px;
  font-weight: 200;
  letter-spacing: 2px;
}

.weather__key {/*styling the weather__key class*/
  color: darkblue;
  border-bottom: solid 2px rgba(255,255,255,0.06);
  padding: 20px 0 20px 0;
  font-weight: 400;
}

.weather__key:last-child {/*styling the weather__key class*/
  border: 0;
}

.weather__value {/*styling the weather__value class*/
  color: green;
  font-weight: 200;
}

.weather__error {/*styling the weather__error class*/
  color: #fff;
  font-size: 20px;
  letter-spacing: 1px;
  font-weight: 200;
}