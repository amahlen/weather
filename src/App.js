import React from "react";

//importing the components from compoents folder
import Weather from "./components/weather";
import Form from "./components/form";
import Titles from "./components/title";

//constructing the Api key
const Api_Key = "8d2de98e089f1c28e1a22fc19a24ef04";


//creating a class and extends with React.Component
class App extends React.Component {

//
  state = {//making states

    temperature: undefined,
    city: undefined,
    country: undefined,
    humidity: undefined,
    description: undefined,
    error: undefined
  }

  getWeather = async (e) => {

    const city = e.target.elements.city.value;//constructing a city
    const country = e.target.elements.country.value;//constructing a country
    e.preventDefault();   
    const api_call = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&appid=${Api_Key}`);
    const response = await api_call.json();//constructing a response
    console.log(response);//consoling the response

    if(city && country){//creating and if statement
      this.setState({
        temperature: response.main.temp,
        city: response.name,
        country: response.sys.country,
        humidity: response.main.humidity,
        description: response.weather[0].description,
        error: ""
      })
    }else{
      this.setState({//if the city or country is incorrect or empty there will be and error.
        error: "Please input search values..."
      })
    }
  }

  render() {//rendering the class

    return (//returning the components

      <div>
         <div className="wrapper">
          <div className="main">
            <div className="container">
              <div className="row">
                <div className="col-xs-5 title-container">
                <Titles />
                </div>
                <div className="col-xs-7 form-container">
                <Form loadWeather={this.getWeather} />
                  <Weather
                    temperature={this.state.temperature}
                    city={this.state.city}
                    country={this.state.country}
                    humidity={this.state.humidity}
                    description={this.state.description}
                    error={this.state.error}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>

    )
  }
}
export default App;//exporting the App