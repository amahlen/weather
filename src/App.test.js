import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('looking for a fetch()', done => {
  function callback(data) {
    expect(data).toBe('looking for a fetch()');
    done();
  }

  fetchData(callback);
});
 
 