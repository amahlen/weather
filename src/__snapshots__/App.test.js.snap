// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`succesfull matches the snapshot 1`] = `
<form>
  <input
    name="city"
    placeholder="City..."
    type="text"
  />
  <input
    name="country"
    placeholder="Country..."
    type="text"
  />
  <button>
    Get Weather
  </button>
</form>
`;
